numpy/testing/_private/utils.py

```diff
@@ -695,9 +695,14 @@ def isnumber(x):
     def istime(x):
         return x.dtype.char in "Mm"
 
-    def chk_same_position(x_id, y_id, hasval='nan'):
-        """Handling nan/inf: check that x and y have the nan/inf at the same
-        locations."""
+    def func_same_pos(func, x, y, hasval='nan'):
```

Comment:
Needs a full docstring. I think the original name is a bit better. Maybe change the signature to `(x, y, func, hasval='nan')`. Do we really need both `hasval` and `func`?

Reply:
`func` does not always have a name (used with a `lambda` below to test for +/- inf). I'll change to `func_chk_same_pos`.